dockers/ubuntu-14.04/.dockerignore

```diff
@@ -0,0 +1,2 @@
+*
+!start_azure.sh
```

Comment:
adding a `.dockerignore` isn't strictly necessary to add support  for mamba, but I think it's generally a good idea and thought now was a good time, given that we're about to test this new image anyway.

For details on this file, you can see https://codefresh.io/docker-tutorial/not-ignore-dockerignore-2/.

Benefits:

* smaller build context = faster builds
* reduced risk of accidentally including unnecessary files
    - smaller images
    - less risk of including secret / sensitive information